chia/wallet/trading/offer.py

```diff
@@ -148,6 +150,40 @@ def __post_init__(self) -> None:
             if max_cost < 0:
                 raise ValidationError(Err.BLOCK_COST_EXCEEDS_MAX, "compute_additions for CoinSpend")
         object.__setattr__(self, "_additions", adds)
+        object.__setattr__(self, "_conditions", None)
+
+    def conditions(self) -> Dict[Coin, List[Condition]]:
+        if self._conditions is None:
+            conditions: Dict[Coin, List[Condition]] = {}
+            max_cost = DEFAULT_CONSTANTS.MAX_BLOCK_COST_CLVM
+            for cs in self._bundle.coin_spends:
+                try:
+                    cost, conds = cs.puzzle_reveal.run_with_cost(max_cost, cs.solution)
+                    max_cost -= cost
+                    conditions[cs.coin] = parse_conditions_non_consensus(conds.as_iter())
+                except Exception:  # pragma: no cover
+                    continue
```

Comment:
I suppose getting these conditions is best-effort. I can imagine spending a long time trouble-shooting an issue, to then find that this throws without leaving a trace

Reply:
Yeah this mimics the additions logic above.  Some offers raise (which is potentially not great and will be fixed in the future) and we need to ignore those ones.